### ISSUE TITLE
Implement backpropagation

### ISSUE THREAD

**Issue by aram-father** [aram-father/mnist-etude#1]
Implement backpropagation for SGD(Stochastic Gradient Descent)
